sw/device/tests/consecutive_irqs/consecutive_irqs_test.c

```diff
@@ -0,0 +1,292 @@
+  // Claim the IRQ by reading the Ibex specific CC register.
+  dif_irq_claim_data_t claim_data;
+  if (!dif_plic_irq_claim(&plic0, PLIC_TARGET, &claim_data)) {
+    debug_msg_and_abort("HANDLER: ISR is not implemented!\n");
+  }
+
+  // Check if the interrupted peripheral is UART.
+  if (claim_data.peripheral != kDifPlicPeripheralUart) {
+    debug_msg_and_abort("HANDLER: ISR interrupted peripheral is not UART!\n");
+  }
+  handler_uart_isr(&claim_data);
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  if (!dif_plic_irq_complete(&plic0, &claim_data)) {
+    debug_msg_and_abort("HANDLER: unable to complete the IRQ request!\n");
+  }
+}
+
+static void uart_initialise(mmio_region_t base_addr, dif_uart_t *uart) {
+  dif_uart_config_t config = {
+      .baudrate = UART_BAUD_RATE,
+      .clk_freq_hz = kIbexClockFreqHz,
+      .parity_enable = kDifUartDisable,
+      .parity = kDifUartParityEven,
+  };
+
+  // No debug output in case of UART initialisation failure.
+  if (!dif_uart_init(base_addr, &config, uart)) {
+    abort();
+  }
+}
+
+static bool plic_initialise(mmio_region_t base_addr, dif_plic_t *plic) {
+  if (!dif_plic_init(base_addr, plic)) {
+    print_log(uart_print_char_p, "PLIC: init failed!\n");
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Configures all the relevant interrupts in UART.
+ */
+static bool uart_configure_irqs(dif_uart_t *uart) {
+  // Clear the Interrupt State for all relevant IRQs. These must be cleared
+  // before the IRQs are enabled in PLIC to make that the previous
+  // interrupt state (for example, inherrited from the bootrom) affect the
+  // test. TODO - really a DIF routine is needed to reset the uart to defaults.
+  if (!dif_uart_irq_state_clear(uart, kDifUartInterruptRxOverflow)) {
+    print_log(uart_print_char_p, "UART: failed to clear TX Watermark IRQ!\n");
+    return false;
+  }
+  if (!dif_uart_irq_state_clear(uart, kDifUartInterruptTxEmpty)) {
+    print_log(uart_print_char_p, "UART: failed to clear TX Empty IRQ!\n");
+    return false;
+  }
+
+  if (!dif_uart_irq_enable(&uart0, kDifUartInterruptRxOverflow,
+                           kDifUartEnable)) {
+    print_log(uart_print_char_p, "UART: RX overflow IRQ enable failed!\n");
+    return false;
+  }
+  if (!dif_uart_irq_enable(&uart0, kDifUartInterruptTxEmpty, kDifUartEnable)) {
+    print_log(uart_print_char_p, "UART: TX empty IRQ enable failed!\n");
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Configures all the relevant interrupts in PLIC.
+ */
+static bool plic_configure_irqs(dif_plic_t *plic) {
+  // Set IRQ triggers to be level triggered
+  if (!dif_plic_irq_trigger_type_set(plic, kDifPlicIrqIdUartRxOverflow,
+                                     kDifPlicDisable)) {
+    print_log(uart_print_char_p,
+              "PLIC: RX overflow trigger type set failed!\n");
+    return false;
+  }
+  if (!dif_plic_irq_trigger_type_set(plic, kDifUartInterruptTxEmpty,
```

Comment:
`kDifUartInterruptTxEmpty` is a `dif_uart_interrupt_t` but `dif_plic_irq_trigger_type_set` expects a `dif_plic_irq_id_t`. Is this intentional?

Reply:
@alphan thank you, that is a genuine bug, I will fix it.

Reply:
I have fixed it now, but I think I will push an update after #1520 is resolved.